app/src/components/publish/PublishSurveyDialog.tsx

```diff
@@ -0,0 +1,204 @@
+import yup from 'utils/YupSchema';
+import SelectAllButton from './SelectAllButton';
+import SubmitSection from './SubmitSection';
+
+export interface ISubmitSurvey {
+  unSubmittedObservation: ISurveyObservationData[];
+  unSubmittedSummary: ISurveySummaryData[];
+  unSubmittedReports: IGetSurveyReportAttachment[];
+  unSubmittedAttachments: IGetSurveyAttachment[];
+}
+
+interface IPublishSurveyDialogProps {
+  open: boolean;
+  onClose: () => void;
+}
+
+const surveySubmitFormInitialValues: ISurveySubmitForm = {
+  observations: [],
+  summary: [],
+  reports: [],
+  attachments: []
+};
+
+const surveySubmitFormYupSchema = yup.object().shape({
+  observations: yup.array(),
+  summary: yup.array(),
+  reports: yup.array(),
+  attachments: yup.array()
+});
+
+const excludesArtifactRevisionId = (item: IGetSurveyReportAttachment | IGetSurveyAttachment) => {
+  return !item.supplementaryAttachmentData?.artifact_revision_id;
+};
+
+/**
+ * Survey Publish button.
+ *
+ * @return {*}
+ */
+const PublishSurveyDialog = (props: IPublishSurveyDialogProps) => {
+  const biohubApi = useBiohubApi();
+  const surveyContext = useContext(SurveyContext);
+
+  const {
+    surveyDataLoader,
+    observationDataLoader,
+    artifactDataLoader,
+    summaryDataLoader,
+    surveyId,
+    projectId
+  } = surveyContext;
+
+  const unsubmittedObservations: ISurveyObservationData[] = [];
+  const unsubmittedSummaryResults: ISurveySummaryData[] = [];
+  const unsubmittedReports: IGetSurveyReportAttachment[] =
+    artifactDataLoader.data?.reportAttachmentsList.filter(excludesArtifactRevisionId) ?? [];
+  const unsubmittedAttachments: IGetSurveyAttachment[] =
+    artifactDataLoader.data?.attachmentsList.filter(excludesArtifactRevisionId) ?? [];
+
+  if (
+    observationDataLoader.data?.surveyObservationData &&
+    !observationDataLoader.data.surveyObservationSupplementaryData?.occurrence_submission_id &&
+    observationDataLoader.data.surveyObservationData.status === SUBMISSION_STATUS_TYPE.TEMPLATE_TRANSFORMED
+  ) {
+    unsubmittedObservations.push(observationDataLoader.data.surveyObservationData);
+  }
+
+  if (
+    summaryDataLoader.data?.surveySummaryData &&
+    !summaryDataLoader.data.surveySummarySupplementaryData?.survey_summary_submission_id &&
+    summaryDataLoader.data.surveySummaryData.messages.length === 0
+  ) {
+    unsubmittedSummaryResults.push(summaryDataLoader.data.surveySummaryData);
+  }
+
+  const hasSubmissionData = Boolean(
+    unsubmittedObservations.length > 0 ||
+      unsubmittedSummaryResults.length > 0 ||
+      unsubmittedReports.length > 0 ||
+      unsubmittedAttachments.length > 0
+  );
+
+  return (
+    <>
+      <SubmitBiohubDialog<ISurveySubmitForm>
+        dialogTitle={SubmitSurveyBiohubI18N.submitSurveyBiohubDialogTitle}
+        submissionSuccessDialogTitle={SubmitSurveyBiohubI18N.submitSurveyBiohubSuccessDialogTitle}
+        submissionSuccessDialogText={SubmitSurveyBiohubI18N.submitSurveyBiohubSuccessDialogTitle}
```

Comment:
nit: setting text prop to the title string (same for the survey publish dialog)

Reply:
good catch! Fixed